core-services/pdc/src/main/java/org/silverpeas/core/pdc/pdc/service/PdcSearchProcessor.java

```diff
@@ -0,0 +1,149 @@
+  }
+
+  private List<GlobalSilverContent> getGlobalSilverContents(List<Integer> silverContentIds,
+      String userId) throws ContentManagerException {
+
+    List<GlobalSilverContent> silverContents = new ArrayList<>();
+    if (silverContentIds == null || silverContentIds.isEmpty()) {
+      return silverContents;
+    }
+
+    ContentManager contentManager = new ContentManager();
+    List<String> instanceIds = contentManager.getInstanceId(silverContentIds);
+
+    for (String instanceId : instanceIds) {
+      try {
+        // On récupère tous les silverContentId d'un instanceId
+        List<Integer> allSilverContentIds =
+            contentManager.getSilverContentIdByInstanceId(instanceId);
+
+        // une fois les SilverContentId de l'instanceId récupérés, on ne garde que ceux qui sont
+        // dans la liste résultat (alSilverContentIds).
+        allSilverContentIds.retainAll(silverContentIds);
+
+        ContentPeas contentP = contentManager.getContentPeas(instanceId);
+        if (contentP != null) {
+          // we are going to search only SilverContent of this instanceId
+          ContentInterface contentInterface = contentP.getContentInterface();
+          List<SilverContentInterface> silverContentTempo =
+              contentInterface.getSilverContentById(allSilverContentIds, instanceId, userId);
+
+          if (silverContentTempo != null) {
+            silverContents.addAll(
+                transformSilverContentsToGlobalSilverContents(silverContentTempo, instanceId));
+          }
+        }
+      } catch (Exception e) {
+        SilverLogger.getLogger(this).error("Can't retrieve content from taxonomy for component {0}",
+            new String[] {instanceId}, e);
+      }
+    }
+    return silverContents;
+  }
+
+  private List<GlobalSilverContent> transformSilverContentsToGlobalSilverContents(
+      List<SilverContentInterface> silverContentTempo, String instanceId) {
+    List<GlobalSilverContent> alSilverContents = new ArrayList<>(silverContentTempo.size());
+    String contentProcessorPrefixId = "default";
+    if (instanceId.startsWith("gallery")) {
+      contentProcessorPrefixId = "gallery";
+    } else if (instanceId.startsWith("kmelia")) {
+      contentProcessorPrefixId = "kmelia";
+    }
+    IGlobalSilverContentProcessor processor =
+        ServiceProvider.getService(contentProcessorPrefixId + PROCESSOR_NAME_SUFFIX);
+
+    for (SilverContentInterface sci : silverContentTempo) {
+      GlobalSilverContent gsc = processor.getGlobalSilverContent(sci);
+      alSilverContents.add(gsc);
+    }
+    return alSilverContents;
+  }
+
+  private List<SearchResult> toSearchResults(List<GlobalSilverContent> contents) {
+    List<SearchResult> results = new ArrayList<>();
+    for (GlobalSilverContent gsc : contents) {
+      results.add(SearchResult.fromGlobalSilverContent(gsc));
+    }
+    return results;
+  }
+
+  private Comparator<GlobalSilverContent> cDateDesc = (o1, o2) -> {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this variable to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1213)